hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactUtil.java

```diff
@@ -165,4 +173,44 @@ public static Short[] characterArrayAsShortArray(Character[] characterArray) {
         }
         return shortArray;
     }
+
+    public static void verifyClassIsCompactSerializable(Class<?> clazz) {
+        if (canBeSerializedAsCompact(clazz)) {
+            return;
+        }
+
+        throw new HazelcastSerializationException("The '" + clazz + "' cannot "
+                + "be serialized with zero configuration Compact serialization "
+                + "because this type is not supported yet. If you want to "
+                + "serialize this class, consider writing a CompactSerializer "
+                + "for it.");
+    }
+
+    public static void verifyFieldClassIsCompactSerializable(Class<?> fieldClass, Class<?> clazz) {
+        if (canBeSerializedAsCompact(fieldClass)) {
+            return;
+        }
+
+        throw new HazelcastSerializationException("The '" + fieldClass + "' "
+                + "cannot be serialized with zero configuration Compact "
+                + "serialization because this type is not supported yet. If you "
+                + "want to serialize '" + clazz + "' which uses this class in "
+                + "its fields, consider writing a CompactSerializer for it.");
+    }
+
+    private static boolean canBeSerializedAsCompact(Class<?> clazz) {
+        Package classPackage = clazz.getPackage();
+        if (classPackage == null) {
+            return false;
+        }
+
+        String name = classPackage.getName();
+        for (String suffix : FORBIDDEN_PACKAGE_NAME_SUFFIXES) {
+            if (name.startsWith(suffix)) {
```

Comment:
Do you want exact match or prefix matching?

Reply:
Good question, I have improved the matching logic in https://github.com/hazelcast/hazelcast/pull/21918/commits/c1d52571cf67f462dd5fdf65398adce5461ea410

For package prefix like `java`, we should be checking for prefix matching with `java.` or exact matching with `java`